oi/FuncGen.cpp

```diff
@@ -391,3 +430,217 @@ void FuncGen::DeclareGetContainer(std::string& testCode) {
       )";
   testCode.append(func);
 }
+
+void FuncGen::DefineDataSegmentDataBuffer(std::string& testCode) {
+  constexpr std::string_view func = R"(
+    namespace ObjectIntrospection {
+    namespace DataBuffer {
+    class DataBuffer {
+      protected:
+        void write_byte(uint8_t);
+    };
+    class DataSegment: public DataBuffer {
+      public:
+        DataSegment(size_t offset) : buf(dataBase + offset) {}
+        void write_byte(uint8_t byte) {
+          if (buf > (dataBase + dataSize * 8)) {
+            // TODO: fail
```

Comment:
TODO: check how the existing method fails and replicate that

Reply:
Apparently the current method keeps moving the end of the buffer but stops writing anything... I'll replicate that.